compiler/daml-lf-verify/BUILD.bazel

```diff
@@ -0,0 +1,113 @@
+# Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+load("//bazel_tools:haskell.bzl", "da_haskell_binary", "da_haskell_library", "da_haskell_test")
+
+da_haskell_binary(
+    name = "daml-lf-verify",
+    srcs = glob(["src/**/*.hs"]),
+    data = [
+        "@z3_nix//:bin/z3",
+    ],
+    hackage_deps = [
+        "aeson",
+        "base",
+        "bytestring",
+        "containers",
+        "deepseq",
+        "Decimal",
+        "extra",
+        "filepath",
+        "ghc-lib-parser",
+        "hashable",
+        "lens",
+        "mtl",
+        "optparse-applicative",
+        "recursion-schemes",
+        "safe",
+        "scientific",
+        "simple-smt",
+        "template-haskell",
+        "text",
+        "time",
+        "unordered-containers",
+        "zip-archive",
+    ],
+    main_function = "DA.Daml.LF.Verify.main",
+    src_strip_prefix = "src",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//compiler/daml-lf-ast",
+        "//compiler/daml-lf-proto",
+        "//compiler/daml-lf-reader",
+        "//compiler/daml-lf-tools",
+        "//libs-haskell/bazel-runfiles",
+        "//libs-haskell/da-hs-base",
+    ],
+)
+
+da_haskell_library(
+    name = "daml-lf-verify-lib",
+    srcs = glob(["src/**/*.hs"]),
+    data = [
+        "@z3_nix//:bin/z3",
+    ],
+    hackage_deps = [
+        "aeson",
+        "base",
+        "bytestring",
+        "containers",
+        "deepseq",
+        "Decimal",
+        "extra",
+        "filepath",
+        "ghc-lib-parser",
+        "hashable",
+        "lens",
+        "mtl",
+        "optparse-applicative",
+        "recursion-schemes",
+        "safe",
+        "scientific",
+        "simple-smt",
+        "template-haskell",
+        "text",
+        "time",
+        "unordered-containers",
+        "zip-archive",
+    ],
+    src_strip_prefix = "src",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//compiler/daml-lf-ast",
+        "//compiler/daml-lf-proto",
+        "//compiler/daml-lf-reader",
+        "//compiler/daml-lf-tools",
+        "//libs-haskell/bazel-runfiles",
+        "//libs-haskell/da-hs-base",
+    ],
+)
+
+da_haskell_test(
+    name = "verify-tests",
+    srcs = glob(["tests/**/*.hs"]),
+    data = glob(["data/*.dar"]),
```

Comment:
We should not check in the DAR files directly but rather the DAML source. You can use the `daml_compile` rule to produce the DARs then. You can find an example at https://github.com/digital-asset/daml/blob/9b476416b8b16ee936e10cc97f11dd1461f8b173/language-support/ts/codegen/tests/BUILD.bazel#L12.